back-end/API/app/services/auth_service.py

```
@@ -0,0 +1,74 @@
+logger = getLogger(__name__)
+
+hasher = PasswordHasher()
+
+class AuthService:
+    def login(self, dto: UserLoginDTO):
+        try:
+            user = user_repo.find_by_email(dto.email)
+            if not user.active:
+                raise LoginException(True, f"Impossible de se connecter avec l'email: {dto.email}")
+            if not hasher.verify(user.password, dto.password):
+                raise LoginException(f"Login attempt failed on user: {dto.email} invalid password")
```

Ruff (0.15.13)

[warning] Abstract `raise` to an inner function

(TRY301)